agents/labjackT7

```diff
@@ -0,0 +1,215 @@
+from ocs.ocs_twisted import TimeoutLock
+
+from autobahn.wamp.exception import ApplicationError
+
+
+'''
+Convert Data
+'''
+
+def float2int(num):
+    return struct.unpack("=i", struct.pack("=f", num))[0]
+
+
+def concatData(data):
+    tVal = 0
+    upper = True
+    for reg in data:
+        if upper:
+            tVal = ((reg & 0xFFFF) << 16)
+            upper = False
+        else:
+            tVal = tVal | (reg & 0xFFFF)
+            upper = True
+    return tVal
+
+
+'''
+Converting numbers to 16-bit data arrays
+'''
+
+def uint16_to_data(num):
+    return struct.unpack("=H", struct.pack("=H", num & 0xFFFF))[0]
+
+
+def uint32_to_data(num):
+    data = [0, 0]
+    data[0] = struct.unpack("=H", struct.pack("=H", (num>>16)&0xffff))[0]
+    data[1] = struct.unpack("=H", struct.pack("=H", num&0xffff))[0]
+    return data
+
+
+def int32_to_data(num):
+    data = [0, 0]
+    data[0] = struct.unpack("=H", struct.pack("=H", (num >> 16) & 0xffff))[0]
+    data[1] = struct.unpack("=H", struct.pack("=H", num & 0xffff))[0]
+    return data
+
+
+def float32_to_data(num):
+    intNum = float2int(num)
+    data = [0, 0]
+    data[0] = (intNum >> 16) & 0xFFFF
+    data[1] = intNum & 0xFFFF
+    return data
+
+
+'''
+Converting data arrays to numbers
+'''
+
+def data_to_uint16(data):
+    return data[0]
+
+
+def data_to_uint32(data):
+    return concatData(data)
+
+
+def data_to_int32(data):
+    return struct.unpack("=i", struct.pack("=I", concatData(data)))[0]
+
+
+def data_to_float32(data):
+    return struct.unpack("=f", struct.pack("=I", concatData(data)))[0]
+
+
+
+'''
+LabJack agent class
+'''
+
+class LabJackT7_agent:
+    def __init__(self,agent,ip_address,num_channels):
+        self.active = True
+        self.agent = agent
+        self.ip_address = ip_address
+        self.module = None
+        self.log = agent.log
+        self.lock = TimeoutLock()
+        self.log = agent.log
+        self.sensors = ['Channel {}'.format(i + 1) for i in range(num_channels)]
+        
+        self.initialized = False
+        self.take_data = False
+        
+        # Register feed
+        agg_params = {
+            'frame_length': 60,
+        }
+        self.agent.register_feed('Sensors',
+                                 record=True,
+                                 agg_params=agg_params,
+                                 buffer_time=1)
+
+
+    def init_labjack_task(self, session, params = None):
+        """
+        task to initialize labjack module
+        """
+        
+        if self.initialized:
+            return True, "Already initialized module"
+        
+        with self.lock.acquire_timeout(0, job = 'init') as acquired:
+            if not acquired:
+                self.log.warn("Could not start init because" "{} is already running".format(self.lock.job))
+                return False, "Could not acquire lock."
+    
+            session.set_status('starting')
+        
+        self.module = ModbusTcpClient(str(self.ip_address))
```

Comment:
This should also be within the "with lock as acquired" block, else it'll always run, even if the lock can't be acquired.